src/Nlog.RabbitMQ.Target/MessageFormatter.cs

```diff
@@ -47,7 +47,20 @@ public static string GetMessageInner(JsonSerializer jsonSerializer, string mesag
 
                     logLine.AddField(key, propertyPair.Value);
                 }
-            }
+            }
+			
+            if (addGdc)
+                foreach (var gdc in GlobalDiagnosticsContext.GetNames())
+                    logLine.AddField(gdc, GlobalDiagnosticsContext.GetObject(gdc));
+
+            if (addNdlc)
+                foreach (var ndlc in NestedDiagnosticsLogicalContext.GetAllObjects())
+                    logLine.AddField("nested", ndlc);
```

Comment:
This will not work when RabbitMQTarget  is used together with `<targets async="true">`. This is why `TargetWithContext` was introduced.

Reply:
It seems not to be included in GetAllProperties(logEvent) even with includeNdlc is true on TargetWithContext.

Reply:
Ndlc is special because there is no key-value-pair only "scope-objects". You can get the scope-objects by calling `GetContextNdlc()`. Think you should drop the Ndlc-part, and instead just tell users to use `<field key="nested" layout="${ndlc}" />`

Reply:
The property IncludeNdlc is already in TargetWithContext, so I prefer use it and automatically add a keys "nested" or "ndlc"